### DISCUSSION TITLE
EPG period ahead too short

### DISCUSSION THREAD

**Discussion by JanToftdahl** [djp952/pvr.hdhomerundvr#123] [Q&A]
Hi, I am struggling with my EPG in KODI 19.5. I am using this AddOn. It will only show EPG 3 days ahead. I would expect 14 days ahead due to my PVR subscription. Is this an known issue?

**Comment by djp952** [comment]
None that I am aware of or can detect on this end. There are two explanations I can think of. The most likely is that Kodi's EPG "Future Days" setting is set to 3:

![Untitled](https://user-images.githubusercontent.com/706055/211128707-422f40c2-dd9e-4a13-99d4-64d0b6ef2de2.png)

Note that you do have to have the settings level set to "Advanced" or "Expert" (lower left) to see this.  If this is the case, you can increase that (again?) to anything you want, up to 14 days.  The addon is required to download 14 days of data at a time so there is no harm going that high.  Give the addon a minute or so to repopulate the EPG if this is the problem.

The other scenario is that there is something wrong with the DVR subscription, or accessing it, that started happening 11 days ago or so, and the EPG data is slowly bleeding off on you since it's not being refreshed. Couple things we can check to find out.

First, can we look at what this "Client specific" option says, if anything?  Maybe wait a good 3-4 minutes after Kodi has started to give enough time for any error accessing EPG to appear:

![Untitled2](https://user-images.githubusercontent.com/706055/211128854-6e124586-25a9-4854-8e20-ac4021feb966.png)

If no errors show up there, let's check this too:

![Untitled3](https://user-images.githubusercontent.com/706055/211128913-22cc86ae-4fef-466c-a35a-4cea2b920f9f.png)

There needs to be at least one device that has "(DVR Authorized)" at the end of the line. If there isn't, we might have to contact SiliconDust for help:

![Untitled4](https://user-images.githubusercontent.com/706055/211128947-87fd00c9-26b2-438c-85f4-d3dc78cf4954.png)

If future days is above 3, there are no error messages, and you have a "(DVR Authorized)" tuner, let me know! The next steps will be to look at the Kodi logs to see if anything odd is going on, but I'd say 99/100 times it's either the Kodi setting or something is off with the DVR subscription :)

I hope it's an easy fix!

**Comment by JanToftdahl** [comment]
Hi Michael,

 �

Your answer was very helpful.

 �

The problem was the “Future days to display” setting.

 �

I was not aware of the different modes for editing the settings.

 �

I changed to “Expert” mode and increased the “Future days to display” from 3 days to 14 days.

 �

Once again thanks for your reply.

 �

Best regards

Jan Toftdahl

 �
